src/DurableTask.AzureStorage/Partitioning/TablePartitionManager.cs

```diff
@@ -0,0 +1,593 @@
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Azure;
+    using Azure.Data.Tables;
+    using DurableTask.AzureStorage.Storage;
+
+    /// <summary>
+    /// Partition ManagerV3 is based on the Table storage.  
+    /// </summary>
+    sealed class TablePartitionManager : IPartitionManager, IDisposable
+    {
+        readonly AzureStorageClient azureStorageClient;
+        readonly AzureStorageOrchestrationService service;
+        readonly AzureStorageOrchestrationServiceSettings settings;
+        readonly CancellationTokenSource partitionManagerCancellationSource;
+        readonly string partitionTableName;
+        readonly string storageAccountName;
+        TableServiceClient tableServiceClient;
+        TableClient partitionTable;
+        readonly TableLeaseManager tableLeaseManager;
+
+        /// <summary>
+        /// constructor to initiate new instances of TablePartitionManager
+        /// </summary>
+        /// <param name="azureStorageClient">Client for the storage account.</param>
+        /// <param name="service">The service responsible for initiating or terminating the partition manager.</param>
+        public TablePartitionManager(
+            AzureStorageOrchestrationService service,
+            AzureStorageClient azureStorageClient)
+        {
+            this.azureStorageClient = azureStorageClient;
+            this.service = service;
+            this.settings = this.azureStorageClient.Settings;
+            this.partitionTableName = this.settings.PartitionTableName;
+            this.storageAccountName = this.azureStorageClient.TableAccountName;
+            this.partitionManagerCancellationSource = new CancellationTokenSource();
+            this.tableServiceClient = new TableServiceClient(this.settings.StorageConnectionString);
+            this.partitionTable = tableServiceClient.GetTableClient(this.partitionTableName);
+            this.tableLeaseManager = new TableLeaseManager(this.partitionTable, this.service, this.settings, this.storageAccountName);
+        }
+
+        
+        /// <summary>
+        /// This method create a new instance of the class TableLeaseManager that represents the worker. 
+        /// And then start the loop that the worker keeps operating on the table. 
+        /// </summary>
+        async Task IPartitionManager.StartAsync()
+        {
+            await Task.Factory.StartNew(() => this.PartitionManagerLoop(this.partitionManagerCancellationSource.Token));
+            this.settings.Logger.PartitionManagerInfo(
+                this.storageAccountName,
+                this.settings.TaskHubName,
+                this.settings.WorkerId,
+                "", //Empty string as it does not target any particular partition, but rather only initiates the partition manager.
+                $"Worker {this.settings.WorkerId} starts working on acquiring and balancing leases.");
```

Comment:
nit: let's simplify this log
```suggestion
                $"Worker {this.settings.WorkerId} starts acquiring and balancing leases.");
```